project/cmake/CPackConfigDEB.cmake

```diff
@@ -0,0 +1,267 @@
+${CPACK_PACKAGE_DESCRIPTION}\n \
+This is the development package for Kodi audio add-ons.")
+
+# kodi-inputstream-dev package
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_FILE_NAME ${CPACK_PACKAGE_NAME}-inputstream-dev_${PACKAGE_NAME_VERSION}_all.deb)
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_PACKAGE_NAME ${CPACK_PACKAGE_NAME}-inputstream-dev)
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_PACKAGE_DEPENDS "kodi-addon-dev")
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_PACKAGE_SECTION libdevel)
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_PACKAGE_PRIORITY optional)
+set(CPACK_DEBIAN_KODI_INPUTSTREAM_DEV_PACKAGE_PROVIDES "xbmc-inputstream-dev")
+set(CPACK_COMPONENT_KODI_INPUTSTREAM_DEV_DESCRIPTION "\
+Kodi Media Center (audio add-on dev package)\n\
+${CPACK_PACKAGE_DESCRIPTION}\n \
+This is the Development package for Kodi inputstream add-ons.")
+
+# kodi-pvr-dev package
+set(CPACK_DEBIAN_KODI_PVR_DEV_FILE_NAME ${CPACK_PACKAGE_NAME}-pvr-dev_${PACKAGE_NAME_VERSION}_all.deb)
+set(CPACK_DEBIAN_KODI_PVR_DEV_PACKAGE_NAME ${CPACK_PACKAGE_NAME}-pvr-dev)
+set(CPACK_DEBIAN_KODI_PVR_DEV_PACKAGE_DEPENDS "kodi-addon-dev")
+set(CPACK_DEBIAN_KODI_PVR_DEV_PACKAGE_SECTION libdevel)
+set(CPACK_DEBIAN_KODI_PVR_DEV_PACKAGE_PRIORITY optional)
+set(CPACK_DEBIAN_KODI_PVR_DEV_PACKAGE_PROVIDES "xbmc-pvr-dev")
+set(CPACK_COMPONENT_KODI_PVR_DEV_DESCRIPTION "\
+Kodi Media Center (PVR add-on dev package)\n\
+${CPACK_PACKAGE_DESCRIPTION}\n \
+This is the Development package for Kodi PVR add-ons.")
+
+# kodi-screensaver-dev package
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_FILE_NAME ${CPACK_PACKAGE_NAME}-screensaver-dev_${PACKAGE_NAME_VERSION}_all.deb)
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_PACKAGE_NAME ${CPACK_PACKAGE_NAME}-screensaver-dev)
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_PACKAGE_DEPENDS "kodi-addon-dev")
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_PACKAGE_SECTION libdevel)
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_PACKAGE_PRIORITY optional)
+set(CPACK_DEBIAN_KODI_SCREENSAVER_DEV_PACKAGE_PROVIDES "xbmc-screensaver-dev")
+set(CPACK_COMPONENT_KODI_SCREENSAVER_DEV_DESCRIPTION "\
+Kodi Media Center (screensaver add-on dev package)\n\
+${CPACK_PACKAGE_DESCRIPTION}\n \
+This is the Development package for Kodi screensaver add-ons.")
+
+# kodi-visualization-dev package
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_FILE_NAME ${CPACK_PACKAGE_NAME}-visualization-dev_${PACKAGE_NAME_VERSION}_all.deb)
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_NAME ${CPACK_PACKAGE_NAME}-visualization-dev)
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_DEPENDS "kodi-addon-dev")
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_SECTION libdevel)
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_PRIORITY optional)
+set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_PROVIDES "xbmc-visualization-dev")
+set(CPACK_COMPONENT_KODI_VISUALIZATION_DEV_DESCRIPTION "\
+Kodi Media Center (visualization add-on dev package)\n\
+${CPACK_PACKAGE_DESCRIPTION}\n \
+This is the Development package for Kodi visualization add-ons.")
```

Comment:
I mean store the metadata in text files, not literary a control file a
such. Costs some more cmake code now, but it means separation of code and
data and in the long term that is always a good precaution to avoid stupid
mistakes where code is changed by accident. The same reasoning is behind a
lot of the buildsystem, paths separated from code etc.
16. jun. 2016 15:56 skrev "Cristiano A. Silva" <notifications@github.com>:

> In project/cmake/CPackConfigDEB.cmake
> <https://github.com/xbmc/xbmc/pull/9987#discussion_r67348291>:
>
> > +set(CPACK_COMPONENT_KODI_SCREENSAVER_DEV_DESCRIPTION "\
> > +Kodi Media Center (screensaver add-on dev package)\n\
> > +${CPACK_PACKAGE_DESCRIPTION}\n \
> > +This is the Development package for Kodi screensaver add-ons.")
> > +
> > +# kodi-visualization-dev package
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_FILE_NAME ${CPACK_PACKAGE_NAME}-visualization-dev_${PACKAGE_NAME_VERSION}_all.deb)
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_NAME ${CPACK_PACKAGE_NAME}-visualization-dev)
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_DEPENDS "kodi-addon-dev")
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_SECTION libdevel)
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_PRIORITY optional)
> > +set(CPACK_DEBIAN_KODI_VISUALIZATION_DEV_PACKAGE_PROVIDES "xbmc-visualization-dev")
> > +set(CPACK_COMPONENT_KODI_VISUALIZATION_DEV_DESCRIPTION "\
> > +Kodi Media Center (visualization add-on dev package)\n\
> > +${CPACK_PACKAGE_DESCRIPTION}\n \
> > +This is the Development package for Kodi visualization add-ons.")
>
> Uhm... are you talking about using
> CPACK_DEBIAN_component_name_PACKAGE_CONTROL_EXTRA with per component
> control file?
>
> If I understand it correctly, the control file is an all or nothing
> approach, i.e. all fields must be populated there.
>
> That brings in another step:
> configure control_component_name.in files to generate a control file per
> component with
> Version: ${PACKAGE_VERSION}, Architecture: ${CPACK_SYSTEM_NAME} and
> perhaps also Maintainer: ${PACKAGE_MAINTAINER} (or alternatively hardcode
> maintainer) during configuration.
>
> Is that what you mean?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/9987/files/c4d6cbc98e4e9ea2c4a94d9af35aa61498c73b18..11ff2d10a1fe869076b68417ab4c43f8e9f0a56f#r67348291>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFktYBVRaN38N6PLpSbGa1MXZpfywVq8ks5qMVYWgaJpZM4I3Vsx>
> .
>